src/Workspaces/Core/Portable/Remote/RemoteHostClient.cs

```diff
@@ -131,7 +123,18 @@ public void Dispose()
 
                 _disposed = true;
 
-                OnDisposed();
+                Dispose(disposing: true);
+                GC.SuppressFinalize(this);
+            }
+
+            ~Connection()
```

Comment:
❕ If this is debug only, the finalizer itself needs to be wrapped in `#if DEBUG` to avoid GC overhead.

Reply:
sure.